tests/attestation_plugin_cert/host/host.cpp

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <limits.h>
+#include <openenclave/attestation/verifier.h>
+#include <openenclave/host.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/tests.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "tls_u.h"
+
+#if defined(_WIN32)
+#include <ShlObj.h>
+#include <Windows.h>
+#endif
+
+#define TEST_EC_KEY 0
+#define TEST_RSA_KEY 1
+#define SKIP_RETURN_CODE 2
+
+#define FILENAME_LENGTH 80
+
+// This is the claims validation callback. A TLS connecting party (client or
+// server) can verify the passed in claims to decide whether to
+// accept a connection request
+oe_result_t enclave_claims_verifier(
+    oe_claim_t* claims,
+    size_t claims_length,
+    void* arg)
+{
+    oe_result_t result = OE_VERIFY_FAILED;
+
+    (void)arg;
+    OE_TRACE_INFO("enclave_claims_verifier is called with claims:\n");
+
+    for (size_t i = 0; i < claims_length; i++)
+    {
+        oe_claim_t* claim = &claims[i];
+        if (strcmp(claim->name, OE_CLAIM_SECURITY_VERSION) == 0)
+        {
+            uint32_t security_version = *(uint32_t*)(claim->value);
+            // Check the enclave's security version
+            if (security_version < 1)
+            {
+                OE_TRACE_ERROR(
+                    "identity->security_version checking failed (%d)\n",
+                    security_version);
+                goto done;
+            }
+        }
+        // Dump an enclave's unique ID, signer ID and Product ID. They are
+        // MRENCLAVE, MRSIGNER and ISVPRODID for SGX enclaves. In a real
+        // scenario, custom id checking should be done here
+        else if (
+            strcmp(claim->name, OE_CLAIM_SIGNER_ID) == 0 ||
+            strcmp(claim->name, OE_CLAIM_UNIQUE_ID) == 0 ||
+            strcmp(claim->name, OE_CLAIM_PRODUCT_ID) == 0)
```

Comment:
Same comment here about separating SGX-specific code from TEE_agnostic code, by moving SGX-only code into a separate file with sgx in the filename or directory name.